packages/mui-utils/src/useForkRef.ts

```diff
@@ -1,22 +1,24 @@
 import * as React from 'react';
 import setRef from './setRef';
 
-export default function useForkRef<InstanceA, InstanceB>(
-  refA: React.Ref<InstanceA> | null | undefined,
-  refB: React.Ref<InstanceB> | null | undefined,
-): React.Ref<InstanceA & InstanceB> | null {
+export default function useForkRef<Instance>(
+  ...refs: Array<React.Ref<Instance> | undefined>
+): React.RefCallback<Instance> | null {
   /**
-   * This will create a new function if the ref props change and are defined.
+   * This will create a new function if the refs passed to this hook change and are all defined.
    * This means react will call the old forkRef with `null` and the new forkRef
    * with the ref. Cleanup naturally emerges from this behavior.
    */
   return React.useMemo(() => {
-    if (refA == null && refB == null) {
+    if (refs.every((ref) => ref == null)) {
       return null;
     }
-    return (refValue) => {
-      setRef(refA, refValue);
-      setRef(refB, refValue);
+
+    return (instance) => {
+      refs.forEach((ref) => {
+        setRef(ref, instance);
+      });
     };
-  }, [refA, refB]);
+    // eslint-disable-next-line react-hooks/exhaustive-deps
```

Comment:
Can you explain a bit more why it is needed even though the `refs` is used as dependencies? Is it because eslint can't detect this use case?

The implementation looks good to me.

Reply:
This ESLint rule requires an array literal to be passed in as it can't verify the correctness of the code otherwise.